File: frontend/src/types/future-letter.ts
/**
 * 未来信状态枚举
 */
export type FutureLetterStatus = 
  | 'scheduled'   // 已安排
  | 'sent'        // 已发送
  | 'cancelled'   // 已取消

/**
 * 投递方式枚举
 */
export type DeliveryMethod = 
  | 'system'      // 系统投递
  | 'courier'     // 信使投递

/**
 * 未来信
 */
export interface FutureLetter {
  id: string
  letter_id: string
  sender_id: string
  recipient_id?: string
  recipient_op_code?: string
  status: FutureLetterStatus
  scheduled_date: Date
  delivery_method: DeliveryMethod
  reminder_enabled: boolean
  reminder_days: number
  last_reminder_sent?: Date
  sent_at?: Date
  created_at: Date
  updated_at: Date

  // 关联对象
  letter?: {
    id: string
    title?: string
    content: string
    style: string
    created_at: Date
  }
  sender?: {
    id: string
    nickname?: string
    avatar_url?: string
  }
  recipient?: {
    id: string
    nickname?: string
    avatar_url?: string
    op_code?: string
  }
}

/**
 * 创建未来信请求
 */
export interface CreateFutureLetterRequest {
  letter_id: string
  scheduled_date: Date
  recipient_id?: string
  recipient_op_code?: string
  delivery_method?: DeliveryMethod
  reminder_enabled?: boolean
  reminder_days?: number
}

/**
 * 未来信列表参数
 */
export interface FutureLetterListParams {
  page?: number
  limit?: number
  status?: FutureLetterStatus
  delivery_method?: DeliveryMethod
  date_from?: string
  date_to?: string
  sort_by?: 'created_at' | 'scheduled_date' | 'sent_at'
  sort_order?: 'asc' | 'desc'
}

/**
 * 未来信统计信息
 */
export interface FutureLetterStats {
  pending_count: number        // 待发送数量
  upcoming_24h_count: number   // 24小时内即将发送数量
  total_system_pending: number // 系统总待发送数量
  sent_count?: number          // 已发送数量
  cancelled_count?: number     // 已取消数量
  total_count: number          // 总计数量
}

/**
 * 更新未来信请求
 */
export interface UpdateFutureLetterRequest {
  scheduled_date?: Date
  delivery_method?: DeliveryMethod
  reminder_enabled?: boolean
  reminder_days?: number
}

/**
 * 未来信响应
 */
export interface FutureLetterResponse {
  id: string
  letter: {
    id: string
    title?: string
    content: string
    style: string
    created_at: Date
  }
  status: FutureLetterStatus
  scheduled_date: Date
  delivery_method: DeliveryMethod
  reminder_enabled: boolean
  reminder_days: number
  last_reminder_sent?: Date
  sent_at?: Date
  recipient?: {
    id: string
    nickname?: string
    avatar_url?: string
    op_code?: string
  }
  created_at: Date
}

/**
 * 未来信提醒设置
 */
export interface ReminderSettings {
  enabled: boolean
  days_before: number
  notification_types: ('email' | 'push' | 'sms')[]
}

/**
 * 未来信日历事件
 */
export interface FutureLetterCalendarEvent {
  id: string
  title: string
  scheduled_date: Date
  status: FutureLetterStatus
  recipient_name?: string
  preview_content: string
}

File: backend/internal/models/future_letter.go
package models

import (
	"time"
	"gorm.io/gorm"
)

type FutureLetterStatus string

const (
	FutureLetterStatusScheduled FutureLetterStatus = "scheduled" // 已计划
	FutureLetterStatusSent      FutureLetterStatus = "sent"      // 已发送
	FutureLetterStatusCancelled FutureLetterStatus = "cancelled" // 已取消
)

type FutureLetter struct {
	ID               string             `json:"id" gorm:"primaryKey;type:varchar(36)"`
	LetterID         string             `json:"letter_id" gorm:"type:varchar(36);not null;uniqueIndex"`
	SenderID         string             `json:"sender_id" gorm:"type:varchar(36);not null;index"`
	RecipientID      string             `json:"recipient_id,omitempty" gorm:"type:varchar(36);index"`
	RecipientOPCode  string             `json:"recipient_op_code,omitempty" gorm:"type:varchar(6);index"`
	Status           FutureLetterStatus `json:"status" gorm:"type:varchar(20);not null;default:'scheduled'"`
	ScheduledDate    time.Time          `json:"scheduled_date" gorm:"not null;index"`
	DeliveryMethod   string             `json:"delivery_method" gorm:"type:varchar(20);default:'system'"` // system/courier
	ReminderEnabled  bool               `json:"reminder_enabled" gorm:"default:true"`
	ReminderDays     int                `json:"reminder_days" gorm:"default:7"` // 提前提醒天数
	LastReminderSent *time.Time         `json:"last_reminder_sent,omitempty"`
	SentAt           *time.Time         `json:"sent_at,omitempty"`
	CreatedAt        time.Time          `json:"created_at"`
	UpdatedAt        time.Time          `json:"updated_at"`
	DeletedAt        gorm.DeletedAt     `json:"-" gorm:"index"`
	
	// 关联
	Letter    *Letter `json:"letter,omitempty" gorm:"foreignKey:LetterID"`
	Sender    *User   `json:"sender,omitempty" gorm:"foreignKey:SenderID"`
	Recipient *User   `json:"recipient,omitempty" gorm:"foreignKey:RecipientID"`
}

func (FutureLetter) TableName() string {
	return "future_letters"
}